toolchain/sem_ir/file.h

```diff
@@ -226,6 +226,14 @@ class File : public Printable<File> {
 
   auto parse_tree() const -> const Parse::Tree& { return *parse_tree_; }
 
+  auto add_non_owning_library(FunctionId function_info) {
+    non_owning_functions_.push_back(function_info);
+  }
+
+  auto get_non_owning_functions() const -> llvm::SmallVector<FunctionId, 0> {
```

Comment:
This is going to return a copy of the vector. Had you considered using `ArrayRef`, which would also elide the size?